gammapy/datasets/map.py

```diff
@@ -637,11 +637,17 @@ def stack(self, other):
         if self.exposure and other.exposure:
             self.exposure.stack(other.exposure, weights=other.mask_safe_image)
             # TODO: check whether this can be improved e.g. handling this in GTI
+            livetime=0*u.s
+            if "livetime" in self.exposure.meta:
```

Comment:
To simplify the stacking logic and improve the performance, the mask handling of "self" is in `.to_masked()`. The standard use case I think is to stack into an empty dataset and there is no point in re-applying the mask to self for each dataset, that is stacked into it. So the `.stack()` method only needs to handle `other.mask_safe`.